web/src/main/java/com/bazaarvoice/emodb/web/resources/blob/BlobStoreResource1.java

```diff
@@ -88,9 +93,59 @@
+    private final LoadingCache<String, Meter> _purgeTableRequestsByApiKey;
+    private final LoadingCache<String, Meter> _getTableAttributesRequestsByApiKey;
+    private final LoadingCache<String, Meter> _setTableAttributesRequestsByApiKey;
+    private final LoadingCache<String, Meter> _getTableOptionsRequestsByApiKey;
+    private final LoadingCache<String, Meter> _getTableSizeRequestsByApiKey;
+    private final LoadingCache<String, Meter> _getTableMetadataRequestsByApiKey;
+    private final LoadingCache<String, Meter> _getObjectMetadataRequestsByApiKey;
+    private final LoadingCache<String, Meter> _scanMetadataRequestsByApiKey;
+    private final LoadingCache<String, Meter> _getTablePlacementsRequestsByApiKey;
+    private final LoadingCache<String, Meter> _getObjectRequestsByApiKey;
+    private final LoadingCache<String, Meter> _putObjectRequestsByApiKey;
+    private final LoadingCache<String, Meter> _deleteObjectRequestsByApiKey;
+
     public BlobStoreResource1(BlobStore blobStore, Set<String> approvedContentTypes, MetricRegistry metricRegistry) {
         _blobStore = blobStore;
         _approvedContentTypes = approvedContentTypes;
+        _metricRegistry = metricRegistry;
+
+        _listTableRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.listTablesByApiKey");
+        _createTableRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.createTableByApiKey");
+        _dropTableRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.dropTableByApiKey");
+        _purgeTableRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.purgeTableByApiKey");
+        _getTableAttributesRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.getTableAttributesByApiKey");
+        _setTableAttributesRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.setTableAttributesByApiKey");
+        _getTableOptionsRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.getTableOptionsByApiKey");
+        _getTableSizeRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.getTableSizeByApiKey");
+        _getTableMetadataRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.getTableMetadataByApiKey");
+        _getObjectMetadataRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.headByApiKey");
+        _scanMetadataRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.scanMetadataByApiKey");
+        _getTablePlacementsRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.getTablePlacementsByApiKey");
+        _getObjectRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.getByApiKey");
+        _putObjectRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.putByApiKey");
+        _deleteObjectRequestsByApiKey = createMetricCache("bv.emodb.blob.BlobStoreResource1.deleteByApiKey");
```

Comment:
To reduce the verbosity - we can push out bv.emodb.blob.BlobStoreResource1 appending to the createMetricsCache function. The  do a Metricregistry.Name("bv.emodb.blob.BlobStoreResource1", metricName) in the createMetricCache.

Reply:
makes sense, fixed